signer.go

```diff
@@ -0,0 +1,91 @@
+package httpsignatures
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/sha512"
+	"fmt"
+	"hash"
+	"io"
+)
+
+const (
+	sha256Name = "sha256"
+	sha512Name = "sha512"
+	rsaName    = "rsa"
+)
+
+// HashingAlgorithm provides a way to get the hashing function and name.
+type HashingAlgorithm struct {
+	Hash        crypto.Hash
+	Name        string
+	FetchHasher func() hash.Hash
+}
+
+// fetchCryptoHash returns supported hashes.
+func fetchCryptoHash(algo crypto.Hash) (*HashingAlgorithm, error) {
+	//nolint:exhaustive // we only want to include the hashes we support
+	switch algo {
+	case crypto.SHA256:
+		return &HashingAlgorithm{algo, sha256Name, sha256.New}, nil
+	case crypto.SHA512:
+		return &HashingAlgorithm{algo, sha512Name, sha512.New}, nil
+	default:
+		return nil, NewInitialisationError(fmt.Sprintf("crypto hash is unknown or unsupported: %s", algo), nil)
+	}
+}
+
+// Signer interface describes something that can sign content.
```

Comment:
```suggestion
// Signer is the interface that wraps the basic Sign method.

// Sign signs given content using rand as a good source of entropy for blinding the signing operation. It returns the generated signature and any error encountered that caused sign to stop early. Sign must return a non-nil error if it cannot properly generate the requested signature. Sign must not modify the slice content, even temporarily.
```

I used [io.Writer](https://pkg.go.dev/io#Writer) interface docs as a template.